packages/huma-web-shared/src/stellar/hooks/useStellarBorrower.ts

```
@@ -0,0 +1,126 @@
+          borrower: stellarAddress,
+        })
+        const [creditRecord, creditConfig, dueDetail, availableForDrawdown] =
+          await Promise.all([
+            creditHashPromise.then((creditHash) =>
+              creditStorageClient.get_credit_record({
+                credit_hash: Buffer.from(creditHash.result),
+              }),
+            ),
+            creditHashPromise.then((creditHash) =>
+              creditStorageClient.get_credit_config({
+                credit_hash: Buffer.from(creditHash.result),
+              }),
+            ),
+            creditHashPromise.then((creditHash) =>
+              creditStorageClient.get_due_detail({
+                credit_hash: Buffer.from(creditHash.result),
+              }),
+            ),
+            poolCreditClient.get_amt_available_for_drawdown({
+              borrower: stellarAddress,
+            }),
+          ])
+        const creditRecordVal = creditRecord.result
+        const creditConfigVal = creditConfig.result
+        const dueDetailVal = dueDetail.result
+        const availableForDrawdownVal = availableForDrawdown.result
+        setCreditRecord(creditRecordVal)
+        setCreditConfig(creditConfigVal)
+        setDueDetail(dueDetailVal)
+
+        const unbilledPrincipalBN = BigInt(creditRecordVal.unbilled_principal)
+        const nextDueBN = BigInt(creditRecordVal.next_due)
+        const yieldDueBN = BigInt(creditRecordVal.yield_due)
+        const totalPastDueBN = BigInt(creditRecordVal.total_past_due)
+        const principalAmountBN =
+          unbilledPrincipalBN + nextDueBN - yieldDueBN + totalPastDueBN
+        setCreditAvailable(availableForDrawdownVal)
+        setPrincipal(principalAmountBN)
+        setTotalDue(
+          BigInt(creditRecordVal.next_due) +
+            BigInt(creditRecordVal.total_past_due),
+        )
+      } catch (error) {
+        console.error('Error fetching Stellar borrower data:', error)
```

Do we need to take action if error happens?

---

We eagerly fetch the stellar borrower info when landing on the pool page, so if this silently fails it either means there was a fetch failure or the wallet is not a borrower of the pool.